internal/test/wmcb/wmcb_test.go

```diff
@@ -182,6 +236,151 @@ func (vm *wmcbVM) remoteDownloadExtract(url, sha, remoteDownloadFile, remoteExtr
+func (vm *wmcbVM) handleHybridOverlay(nodeName string) error {
+	// Check if the hybrid-overlay is running
+	_, stderr, err := vm.Run("Get-Process -Name \"hybrid-overlay\"", true)
+
+	// stderr being empty implies that an hybrid-overlay was running. This is to help with local development.
+	if err == nil || stderr == "" {
+		return nil
+	}
+
+	// Wait until the node object has the hybrid overlay subnet annotation. Otherwise the hybrid-overlay will fail to
+	// start
+	if err = waitForHybridOverlayAnnotation(nodeName); err != nil {
+		return fmt.Errorf("error waiting for hybrid overlay node annotation: %v", err)
+	}
+
+	// Start the hybrid-overlay in the background over ssh. We cannot use vm.Run() and by extension WinRM.Run() here as
+	// we observed WinRM.Run() returning before the commands completes execution. The reason for that is unclear and
+	// requires further investigation.
+	go vm.RunOverSsh(hybridOverlayExecutable+" --node "+nodeName+
+		" --k8s-kubeconfig c:\\k\\kubeconfig > C:\\k\\hybrid-overlay.log 2>&1", false)
+
+	err = vm.waitForHybridOverlayToRun()
+	if err != nil {
+		return fmt.Errorf("error running %s: %v", hybridOverlayName, err)
+	}
+
+	err = vm.waitForOpenShiftHNSNetworks()
+	if err != nil {
+		return fmt.Errorf("error waiting for OpenShift HNS networks to be created: %v", err)
+	}
+
+	// Running the hybrid-overlay causes network reconfiguration in the Windows VM which results in the ssh connection
+	// being closed and the client is not smart enough to . We have observed that the WinRM connection does not get closed. The reason for th
+	err = vm.Reinitialize()
+
+	return nil
+}
+
+// waitForOpenShiftHSNNetworks waits for the OpenShift HNS networks to be created until the timeout is reached
+func (vm *wmcbVM) waitForOpenShiftHNSNetworks() error {
+	hnsNetworksArePresent := false
+	var stdout string
+	var err error
+	for retries := 0; retries < e2ef.RetryCount; retries++ {
+		stdout, _, err = vm.Run("Get-HnsNetwork", true)
+		if err != nil {
+			// retry
+			continue
+		}
+
+		if strings.Contains(stdout, "BaseOpenShiftNetwork") &&
+			strings.Contains(stdout, "OpenShiftNetwork") {
+			hnsNetworksArePresent = true
+			break
+		}
+
+		if !hnsNetworksArePresent {
+			time.Sleep(e2ef.RetryInterval)
+		}
+	}
+
+	if !hnsNetworksArePresent {
+		// Log the HNS networks that are present on the VM
+		log.Printf("Get-HnsNetwork:\n%s", stdout)
+		return fmt.Errorf("timeout waiting for OpenShift HNS networks: %v", err)
+	}
+	return nil
+}
+
+// waitForHybridOverlayToRun waits for the hybrid-overlay.exe to run until the timeout is reached
+func (vm *wmcbVM) waitForHybridOverlayToRun() error {
+	hybridOverlayRunning := false
+	var err error
+	for retries := 0; retries < e2ef.RetryCount; retries++ {
+		_, _, err = vm.Run("Get-Process -Name \"hybrid-overlay\"", true)
+		if err == nil {
+			hybridOverlayRunning = true
+			break
+		}
+
+		if !hybridOverlayRunning {
+			time.Sleep(e2ef.RetryInterval)
```

Comment:
For people unfamiliar, this is a constant back off mechanism. We can have something called exponential back off which ensures we'll wait exponentially after each call.

Reply:
Is this a request to add that in this PR? I suggest doing that in a separate PR as we are doing this in multiple places.

Reply:
Nope, I was suggesting for others to how are unfamiliar.